pyblish_lite/control.py

```diff
@@ -144,36 +202,54 @@ def _process(self, plugin, instance=None):
 
         return result
 
-    def _run(self, until=float("inf"), on_finished=lambda: None):
-        """Process current pair and store next pair for next process
+    def _pair_yielder(self, plugins):
 
-        Arguments:
-            until (pyblish.api.Order, optional): Keep fetching next()
-                until this order, default value is infinity.
-            on_finished (callable, optional): What to do when finishing,
-                defaults to doing nothing.
+        for plugin in plugins:
+            if not plugin.active:
+                pyblish.logic.log.debug("%s was inactive, skipping.." % plugin)
+                continue
 
-        """
+            self.processing["nextOrder"] = plugin.order
 
+            message = self.test(**self.processing)
+            if message:
+                raise pyblish.logic.StopIteration("Stopped due to %s" % message)
+
+            if not self.is_running:
+                raise StopIteration("Stopped")
+
+            if plugin.__instanceEnabled__:
+                instances = pyblish.logic.instances_by_plugin(
+                    self.context, plugin
+                )
+                for instance in instances:
+                    if instance.data.get("publish") is False:
+                        pyblish.logic.log.debug(
+                            "%s was inactive, skipping.." % instance
+                        )
+                        continue
+                    yield plugin, instance
+            else:
+                families = util.collect_families_from_instances(
+                    self.context, only_active=True
+                )
+                plugins = pyblish.logic.plugins_by_families([plugin,], families)
+                if not plugins:
+                    continue
+                yield plugin, None
+
+    def iterate_and_process(
+        self, plugins, on_finished=lambda: None
+    ):
```

Comment:
Minor note, could we put these on one line?

Reply:
done